alica_standard_library/src/GenerateRandom.cpp

```diff
@@ -0,0 +1,31 @@
+#include "GenerateRandom.h"
+
+#include <random>
+
+namespace alica_standard_library
+{
+
+GenerateRandom::GenerateRandom(alica::BehaviourContext& context)
+        : BasicBehaviour(context)
+{
+}
+
+void GenerateRandom::initialiseParameters()
+{
+    alica::LockedBlackboardRW bb(*(getBlackboard()));
```

Comment:
can use `UnlockedBlackboard`

Reply:
oos but I think in situations where it is okay not to lock the blackboard, we should supply it as a function argument as an UnlockedBlackboard already.  I don't think it's a good idea for the user to need to understand the engine well enough to guess when they need to lock or not.  Until that's available I'd kind of prefer to just lock it.  I think we can support this without breaking interface though